src/edu/csus/ecs/pc2/shadow/ShadowController.java

```diff
@@ -345,9 +357,160 @@ private IRemoteContestAPIAdapter createRemoteContestAPIAdapter(URL url, String l
+     * contain the same rank, teamId, number of problems solved, and total (penalty) points).  If for some reason a row exists
+     * in one of the scoreboards but not the other, the corresponding entry in the ShadowScoreboardRowComparison will be null 
+     * (an indication of scoreboards which inherently are not equal).
+     * 
+     * This method is only operative while Shadow operations are running; if shadowing is not running
+     * then the method returns an empty array (that is, an array of size zero).
+     * 
+     * The method also returns an empty array if an error occurs when fetching either the PC2 or the remote CCS 
+     * JSON scoreboard  strings, or when parsing those strings.
+     * 
+     * @return an array of ShadowScoreboardRowComparison information, or an empty array if shadowing isn't running
+     *          or an error occurs during processing either of the JSON strings representing the two scoreboards.
+     */
+    public ShadowScoreboardRowComparison[] getScoreboardComparisonInfo() {
+
+        //initialize return to empty
+        ShadowScoreboardRowComparison[] emptyArray = new ShadowScoreboardRowComparison[0];
+
+        if (getStatus() != SHADOW_CONTROLLER_STATUS.SC_RUNNING) {
+            log.warning("Shadow Controller 'getScoreboardComparisonInfo()' called when Shadow controller is not running");
+            return emptyArray;
+        }
+
+        // get the PC2 scoreboard JSON
+        String pc2Json = getPC2ScoreboardJSON();
+        if (pc2Json==null) {
+            log.warning("Got empty or null JSON scoreboard from PC2");
+            return emptyArray;
+        }
+
+        // get the remote CCS scoreboard JSON
+        String remoteJson = getRemoteScoreboardJSON();
+        if (remoteJson==null) {
+            log.warning("Got empty or null JSON scoreboard from remote CCS");
+            return emptyArray;
+        }
+
+        // construct a comparator to compare the two JSON strings
+        ShadowScoreboardComparisonGenerator comparator = new ShadowScoreboardComparisonGenerator(this);
+
+        //use the comparator to obtain a row-by-row comparison of the two JSON scoreboards
+        ShadowScoreboardRowComparison[] results = comparator.compare(pc2Json, remoteJson);
+
+        return results;
+    }
+
+   /**
+     * This method constructs a PC2 {@link ScoreboardJson} object and uses it to return a String
+     * containing the current PC2 scoreboard in CLICS Contest API format. 
+     * 
+     * @return a String containing the PC2 scoreboard JSON.
+     */
+    private String getPC2ScoreboardJSON() {
+        
+        ScoreboardJson sbJsonObject = new ScoreboardJson();
+        String pc2Json ;
+        try {
+            pc2Json = sbJsonObject.createJSON(localContest, log);
+        } catch (JsonProcessingException | IllegalContestState | JAXBException e) {
+            log.warning("Exception creating PC2 scoreboard JSON: " + e.getMessage());
```

Comment:
May or may not output stack trace to log thus,
log.warning("Exception creating PC2 scoreboard JSON: " + e.getMessage(), e);